apis/bigqueryconnection/v1alpha1/connection_types.go

```diff
@@ -38,8 +38,11 @@ type Parent struct {
 type BigQueryConnectionConnectionSpec struct {
 	Parent `json:",inline"`
 
-	// The BigQuery ConnectionID. This is a server-generated ID in the UUID format.
-	// If not provided, ConfigConnector will create a new Connection and store the UUID in `status.serviceGeneratedID` field.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
+	// Immutable. Optional.
+	// The BigQuery Connection ID used for resource creation or acquisition.
+	// For creation: If specified, this value is used as the connection ID. If not provided, a UUID is generated and stored in the `status.ExternalRef` field.
+	// For acquisition: This field must be provided to identify the connection resource to acquire.
```

Comment:
nit: "otherwise...a UUID is assigned...". We want to make it clear because this use case is uncommon to users, and they do not know the code logic behind.

Reply:
Done.